plugin.rb

```diff
@@ -186,19 +196,58 @@ class ::TopicListItemSerializer
   require_dependency 'list_controller'
   class ::ListController
     generate_message_route(:private_messages_assigned)
-    generate_message_route(:messages_assigned)
   end
 
   add_to_class(:topic_query, :list_messages_assigned) do |user|
-    list = joined_topic_user.where("
+    list = default_results.where("
       topics.id IN (
         SELECT topic_id FROM topic_custom_fields
         WHERE name = 'assigned_to_id'
         AND value = ?)
     ", user.id.to_s)
       .order("topics.bumped_at DESC")
 
-    create_list(:assigned, {}, list)
+    return create_list(:assigned, {}, list)
+  end
+
+  add_to_class(:list_controller, :messages_assigned) do
+    page = (params[:page].to_i || 0).to_i
+
+    user = User.find_by("LOWER(username) = ?", params[:username])
```

Comment:
We have a special column here which is indexed. Use `username_lower` rather than `LOWER(username)`

Or even better, use `User.find_by_username(params[:username])` and it will happen automatically :D